tools/sourcemap-tools/src/SourceProcessor.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs';
+import { BasicSourceMapConsumer, Position, RawSourceMap, SourceMapConsumer, SourceMapGenerator } from 'source-map';
+import { DebugIdGenerator } from './DebugIdGenerator';
+import { stringToUuid } from './helpers/stringToUuid';
+
+export class SourceProcessor {
+    constructor(private readonly _debugIdGenerator: DebugIdGenerator) {}
+
+    public async processSourceAndSourceMap(source: string, sourceMap: string | RawSourceMap, debugId?: string) {
+        if (!debugId) {
+            debugId = stringToUuid(source);
+        }
+
+        const sourceSnippet = this._debugIdGenerator.generateSourceSnippet(debugId);
+        const sourceComment = this._debugIdGenerator.generateSourceComment(debugId);
+
+        const newSource = sourceSnippet + '\n' + source + '\n' + sourceComment;
+
+        // We need to offset the source map by amount of lines that we're inserting to the source code
```

Comment:
because...

Reply:
Sourcemaps map original position to generated position like this:
`original line X, original column Y => generated line A, generated column B`

If I add anything to the generated source, the sourcemap will be invalid, because generated lines will be changed. So if I add a line at the beginning of the file, I need to offset them like this:
`original line X, original column Y => generated line A + 1, generated column B`

Reply:
ah ok got it - so +1 is because of our code injected to each map. Got it. Does it make sense to add it to the comment?

Reply:
I expanded the comment.